pkg/utils/vault/read.go

```diff
@@ -0,0 +1,74 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package vault
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/hashicorp/vault/api"
+)
+
+const (
+	// rootPathEnvVar contains the path to prefix all paths when reading secrets depending on vault backend used.
+	rootPathEnvVar  = "VAULT_ROOT_PATH"
+	defaultRootPath = "secret/ci/elastic-cloud-on-k8s"
+)
+
+// Get fetches contents of a single field at a specified path in Vault
+func (c *Client) Get(secretPath string, fieldName string) (string, error) {
+	result, err := c.GetMany(secretPath, fieldName)
+	if err != nil {
+		return "", err
+	}
+
+	return result[0], nil
```

Comment:
I think `.GetMany` could return a 0 length slice.

Reply:
I don't think it's possible. I will add unit tests.